ci/spelling.sh

```diff
@@ -2,4 +2,4 @@
 
 npm install -g cspell
 git fetch origin main:main
-git diff --name-only main $HEAD | xargs -L1 npx cspell -c ./ci/cspell.json -u -e ci/
+git diff --name-only main $HEAD | grep .md | grep . && xargs -L1 npx cspell -c ./ci/spelling-config.json || echo "no markdown files changed"
```

Comment:
It wasn't obvious to me what `grep .` is doing. I'd rather write this one-liner in another way:
```shell
MD_FILES="$(git diff --name-only main $HEAD | grep '\.md$')"
if [ -n "$MD_FILES" ]; then
    echo "$MD_FILES" | xargs -L1 npx cspell -c ./ci/spelling-config.json
else
    echo "no markdown files changed"
fi
```

Even if you want to leave it as-is, I'd suggest to replace `grep .md` by `grep '\.md$'` Otherwise it matches files like `test.xmd` or `test.md.gz`

Reply:
Oh good call! The If/Else construct definitely helps readability. I'm always so hacky when writing bash. 

We actually updated in https://github.com/cncf/tag-security/commit/38ab2a0bae90b2d77d1e34f8c646bc2de78503c9 so some of this doesn't apply anymore but targeting the .md pattern better in the grep statement is still a great improvement. Do you mind contributing as a change? @lumjjb and myself can review sometime this week.

Reply:
Sure, I've submitted https://github.com/cncf/tag-security/pull/719